From 9be0010e109d307921a049d10078813423227582 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_git@polytechnique.org>
Date: Fri, 15 Dec 2023 18:03:33 +0100
Subject: [PATCH 1/1] Fix compatibility with libxml2 2.12
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Since libxml2 2.12, xmlNodePtr is no longer included by libxml/encoding.h.
This leads to build errors such as:

      CC       reporter_mantisbt-reporter-mantisbt.o
    In file included from reporter-mantisbt.c:22:
    mantisbt.h:48:5: error: unknown type name ‘xmlNodePtr’
       48 |     xmlNodePtr sr_root;
          |     ^~~~~~~~~~
    mantisbt.h:49:5: error: unknown type name ‘xmlNodePtr’
       49 |     xmlNodePtr sr_body;
          |     ^~~~~~~~~~
    mantisbt.h:50:5: error: unknown type name ‘xmlNodePtr’
       50 |     xmlNodePtr sr_method;
          |     ^~~~~~~~~~

Fix this by including libxml/tree.h.

Signed-off-by: Nicolas Iooss <nicolas.iooss_git@polytechnique.org>
---
 src/plugins/mantisbt.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/plugins/mantisbt.h b/src/plugins/mantisbt.h
index 1feaf8f8863e..6979e7244930 100644
--- a/src/plugins/mantisbt.h
+++ b/src/plugins/mantisbt.h
@@ -25,6 +25,7 @@ extern "C" {
 #endif
 
 #include <libxml/encoding.h>
+#include <libxml/tree.h>
 #include "problem_report.h"
 
 #define SOAP_STRING "ns2:string"
-- 
2.42.1

